### PR TITLE
[TEST] Add coverage and fix PermissionError in load_file

### PR DESCRIPTION
**Pull request by RainRat**
Increased reliability and test coverage by handling and testing an edge case (permission errors) during file loading in the `load_file` utility function. Added comprehensive unit tests to ensure this behavior is preserved.

---
*PR created automatically by Jules for task [1067857462877724713](https://jules.google.com/task/1067857462877724713) started by @RainRat*

### CODE REVIEW

File: tests/test_config_and_utils.py [hunk 1]
```diff
@@ -42,6 +42,26 @@ def test_load_file_missing_multi_line():
     result = load_file("non_existent.txt", mode='multi_line')
     assert result == []
 
+def test_load_file_permission_error_single_line(tmp_path):
+    f = tmp_path / "protected.txt"
+    f.write_text("secret")
+    f.chmod(0o000)
+    try:
+        result = load_file(str(f), mode='single_line')
+        assert result == ""
```
[the remark this answers is not in the document]
Comment (RainRat) [reply tests/test_config_and_utils.py:51]: @jules investigate this report
